spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -158,6 +158,428 @@ Make sure to add 'spring-boot-configuration-processor' as an optional dependency
 ----
 ====
 
+[[custom-applications]]
+== Creating custom applications
+
+While there are out of the box source, processor, sink applications available, one can extend these applications or write a custom link:https://github.com/spring-cloud/spring-cloud-stream[Spring Cloud Stream] application.
+
+To create any Spring Cloud Stream application, based on the requirement one or more Spring Cloud Stream binder dependencies need to be added as dependencies for the project.
+
+If there are multiple binders used by the application, refer the instructions <<passing_producer_consumer_properties>> on how to configure them.
+
+In the following section, step by step instructions are provided with an example on how to create custom source, processor and sink applications.
+
+[[custom-source-application]]
+=== Creating custom source
+
+Create a Spring Cloud Stream "Source" application that is responsible for sending the source events into the stream.
```

Comment:
Shouldn't we point them to Initializr ? What's the preferred way of creating apps now ?